packages/nestjs/src/index.ts

```diff
@@ -0,0 +1,340 @@
+import {
+  createGatewayRuntime,
+  GatewayCLIBuiltinPluginConfig,
+  GatewayPlugin,
+  getBuiltinPluginsFromConfig,
+  getCacheInstanceFromConfig,
+  getGraphQLWSOptions,
+  PubSub,
+  type GatewayConfig,
+  type GatewayRuntime,
+} from '@graphql-hive/gateway';
+import {
+  Logger as GatewayLogger,
+  type LazyLoggerMessage,
+} from '@graphql-mesh/types';
+import {
+  asArray,
+  getResolversFromSchema,
+  type IResolvers,
+  type TypeSource,
+} from '@graphql-tools/utils';
+import { Injectable, Logger as NestLogger } from '@nestjs/common';
+import {
+  AbstractGraphQLDriver,
+  GqlSubscriptionService,
+  GraphQLWsSubscriptionsConfig,
+  type GqlModuleOptions,
+  type SubscriptionConfig,
+} from '@nestjs/graphql';
+import type { FastifyReply, FastifyRequest } from 'fastify';
+
+export type HiveGatewayDriverConfig<
+  TContext extends Record<string, any> = Record<string, any>,
+> = GatewayConfig<TContext> &
+  GatewayCLIBuiltinPluginConfig &
+  GqlModuleOptions & {
+    /**
+     * If enabled, "subscriptions-transport-ws" will be automatically registered.
+     */
+    installSubscriptionHandlers?: boolean;
+    /**
+     * Subscriptions configuration.
+     */
+    subscriptions?: SubscriptionConfig;
+  };
+
+@Injectable()
+export class HiveGatewayDriver<
+  TContext extends Record<string, any>,
+> extends AbstractGraphQLDriver<HiveGatewayDriverConfig<TContext>> {
+  private _gatewayRuntime: GatewayRuntime<TContext> | undefined;
+  private _subscriptionService?: GqlSubscriptionService;
```

Comment:
Maybe we can try using [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties#browser_compatibility)?

```suggestion
  #gatewayRuntime: GatewayRuntime<TContext> | undefined;
  #subscriptionService?: GqlSubscriptionService;
```